src/5_0/index.ts

```diff
@@ -0,0 +1,44 @@
+import { createProjectService, PackageJsonAutoImportPreference, ProjectService } from './projectService';
+import { createProject } from './project';
+import type { LanguageService, LanguageServiceHost } from 'typescript/lib/tsserverlibrary';
+
+// only create the once for all hosts, as this will improve performance as the internal cache can be reused
+let projectService: ProjectService;
+
+export default function (ts: typeof import('typescript/lib/tsserverlibrary'), host: LanguageServiceHost, service: LanguageService, rootDirectory: string) {
+	// will need to make this the workspace directory
+	if (!projectService) {
+		projectService = createProjectService(
+			ts, 
+			ts.sys, 
+			rootDirectory,
+			{ includePackageJsonAutoImports: PackageJsonAutoImportPreference.On }, 
+			ts.LanguageServiceMode.Semantic
```

Comment:
Likewise for this; what is the best way to get the current language service mode?

Reply:
We don't need to config this for now, because only the semantic server enabled auto-complete feature.